cmd/utils/flags.go

```diff
@@ -1954,6 +1954,12 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 			cfg.EnableTrustProtocol = true
 		}
 
+    // A node without trie is not able to provide snap data, so it should disable snap protocol.
```

Comment:
if disable
fast node can do snap sync?

Reply:
If you meant when a node starts with a fast node, which does full sync, then later on becomes a full node and does snap sync again, will it work? I'm not entirely sure, my intuition is that it doesn't work.